pootle/apps/pootle_statistics/migrations/0011_cleanup_submissions_squashed_0023_remove_scorelog.py

```diff
@@ -0,0 +1,191 @@
+
+    # assoc state change subs with suggestions and update the sub
+    to_update = []
+    accepted_suggestions = suggestions.filter(state__name="accepted").values(
+        "review_time", "pk", "unit_id", "reviewer_id", "user_id")
+    accept_sugg_state_subs = subs.filter(
+        type=SUGG_ACCEPT_TYPE).filter(field=STATE_FIELD)
+    accept_subs_for_update = accept_sugg_state_subs.select_related(
+        "suggestion").only("id", "type", "submitter_id", "suggestion_id")
+    for suggestion in accepted_suggestions.iterator():
+        suggestion["submitter_id"] = suggestion["reviewer_id"]
+        suggestion["creation_time"] = suggestion["review_time"]
+        suggestion_id = suggestion["pk"]
+        suggestion_user = suggestion["user_id"]
+        del suggestion["review_time"]
+        del suggestion["reviewer_id"]
+        del suggestion["pk"]
+        del suggestion["user_id"]
+        matching_sub = accept_subs_for_update.filter(**suggestion).first()
+        if matching_sub:
+            matching_sub.type = NEW_WEB_TYPE
+            matching_sub.suggestion_id = suggestion_id
+            matching_sub.submitter_id = suggestion_user
+            to_update.append(matching_sub)
+    if to_update:
+        Batch(Submission.objects, batch_size=5000).update(
+            to_update,
+            update_fields=["type", "suggestion_id", "submitter_id"],
+            reduces=False)
+
+    # all remaining valid sugg_accept subs should be field=target and have a suggestion
+    to_update = []
+    accept_sugg_subs = subs.filter(type=SUGG_ACCEPT_TYPE).filter(field=TARGET_FIELD).exclude(suggestion__isnull=True)
+    accept_subs_for_update = accept_sugg_subs.select_related(
+        "suggestion").only("id", "type", "submitter_id", "suggestion__user_id")
+    for sub in accept_subs_for_update.iterator():
+        sub.submitter_id = sub.suggestion.user_id
+        sub.type = NEW_WEB_TYPE
+        to_update.append(sub)
+    if to_update:
+        Batch(Submission.objects, batch_size=5000).update(
+            to_update,
+            update_fields=["type", "submitter_id"],
+            reduces=False)
+
+
+def _cleanup_checks(apps):
+    subs = apps.get_model("pootle_statistics.Submission").objects
+    subs.filter(type=MUTE_TYPE).update(
+        old_value=NEW_UNMUTED,
+        new_value=NEW_MUTED,
+        type=NEW_WEB_TYPE,
+        field=NEW_CHECK_FIELD)
+    subs.filter(type=UNMUTE_TYPE).update(
+        old_value=NEW_MUTED,
+        new_value=NEW_UNMUTED,
+        type=NEW_WEB_TYPE,
+        field=NEW_CHECK_FIELD)
+
+
+def update_subs(apps, schema_editor):
+    _cleanup_checks(apps)
+    _cleanup_accept_suggestion_subs(apps)
+
+
+class Migration(migrations.Migration):
+
+    replaces = [
+        (b'pootle_statistics', '0011_cleanup_submissions'),
+        (b'pootle_statistics', '0012_remove_create_subs'),
+        (b'pootle_statistics', '0013_remove_revert_flag'),
+        (b'pootle_statistics', '0014_truncate_source_subs'),
```

Comment:
this one has been excluded - i raised a pr to also remove the code from that migration because i dont think this was the correct thing to do